src/Libraries/Microsoft.Extensions.Telemetry/Logging/ExtendedLoggerFactory.cs

```diff
@@ -0,0 +1,263 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using Microsoft.Extensions.Compliance.Redaction;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Microsoft.Extensions.Telemetry.Enrichment;
+using Microsoft.Shared.Diagnostics;
+
+namespace Microsoft.Extensions.Telemetry.Logging;
+
+/// <summary>
+/// Produces instances of <see cref="ILogger"/> classes based on the given providers, with support for redaction and enrichment.
+/// </summary>
+[Experimental]
+public sealed class ExtendedLoggerFactory : ILoggerFactory
+{
+    private readonly LoggerFactory _loggerFactory;
+    private readonly ILogEnricher[] _enrichers;
+    private readonly IStaticLogEnricher[] _staticEnrichers;
+    private readonly IRedactorProvider? _redactorProvider;
+    private readonly ConcurrentDictionary<string, ILogger> _cache = new();
+    private readonly IDisposable? _changeTokenRegistration;
+    private ExtendedLoggerFilterOptions _filterOptions;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="ExtendedLoggerFactory"/> class.
+    /// </summary>
+    public ExtendedLoggerFactory()
```

Comment:
Why are there so many constructors.

Reply:
There’s no need to copy all of the constructors as this isn’t a primary public API for consumers to use. It’s just copying the legacy from LoggerFactory and complicating this design